scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestController_SpecificSchedule_Test.java

```diff
@@ -77,23 +80,28 @@ public void beforeTest() throws SchedulerException {
 	@After
 	@Transactional
 	public void afterTest() {
-		for (ScheduleEntity entity : scheduleDao.findAllSchedulesByAppId(appId)) {
-			scheduleDao.delete(entity);
+		for (String appId : multiAppIds) {
+			for (ScheduleEntity entity : scheduleDao.findAllSchedulesByAppId(appId)) {
+				scheduleDao.delete(entity);
+			}
 		}
 	}
 
 	@Test
 	@Transactional
 	public void testGetAllSchedule_with_no_schedules() throws Exception {
-		ResultActions resultActions = callGetAllSchedulesByAppId();
+		ResultActions resultActions = callGetAllSchedulesByAppId(appId);
 		assertSpecificSchedulesFoundEquals(0, resultActions);
 	}
 
 	@Test
 	@Transactional
 	public void testCreateAndGetSchedules() throws Exception {
-		assertCreateAndGetSchedules(1);
-		assertCreateAndGetSchedules(5);
+		assertCreateAndGetSchedules(singleAppId, 1);
+		assertCreateAndGetSchedules(singleAppId, 5);
```

Comment:
Why? Always do the cleanup in a @Before.
Never assume the previous run has done the proper cleanup. And @XXXClass just leaves you open for contamination. Now if you are guaranteeing that all tests are creating separate data, great.